File: packages/backend/src/features/question/question.model.ts
import { QuizzemModel } from '~/src/core/domain/quizzem.model';
import { AnswerModel } from '~/src/features/answer/answer.model';
import { QuestionType } from '~/src/features/question/question-type.model';

export class QuestionModel extends QuizzemModel {
  text: string = '';
  type: QuestionType = 'MULTIPLE_CHOICE';
  correctAnswer: string = '';
  answers: AnswerModel[] = [];
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/question/model/QuestionModel.kt
package quizzem.backend.features.question.model

import quizzem.backend.features.answer.model.AnswerModel
import java.time.Instant
import java.util.*

data class QuestionModel(
    val id: UUID,
    val createdAt: Instant,
    val updatedAt: Instant,

    val text: String = "",
    val type: QuestionType = QuestionType.MULTIPLE_CHOICE,
    val correctAnswer: String = "",
    val answers: List<AnswerModel>
)
